### PR TITLE
Support phi3-vision

### PR DESCRIPTION
**Pull request by RunningLeon**
## Motivation

Support phi3-vision for pytorch engine
- [microsoft/Phi-3-vision-128k-instruct](https://huggingface.co/microsoft/Phi-3-vision-128k-instruct)

as required in #1637 

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repositories?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a dependency on downstream projects of a newer version, this PR should be tested with all supported versions of downstream projects.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: lmdeploy/archs.py [hunk 1]
```diff
@@ -136,6 +136,8 @@ def check_vl_llm(config: dict) -> bool:
         return True
     elif arch == 'LlavaNextForConditionalGeneration':
         return True
+    elif arch == 'Phi3VForCausalLM':
```
Comment (lvhan028) [inline lmdeploy/archs.py:139]: Can we refactor `check_vl_llm`? Checking if `arch` in a list is more user-friendly than adding `elif` one by one

File: lmdeploy/vl/model/phi3_vision.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+            from transformers import AutoModelForCausalLM
+            model = AutoModelForCausalLM.from_config(self.hf_config,
+                                                     trust_remote_code=True)
+            if not self.with_llm:
+                del model.lm_head
+                del model.model.layers
+                del model.model.norm
+                del model.model.embed_tokens
+                del model.model.vision_embed_tokens.wte
+                setattr(model, 'lm_head', None)
```
Comment (lvhan028) [inline lmdeploy/vl/model/phi3_vision.py:146]: Is it necessary after `del model.lm_head`?

Comment (RunningLeon) [reply lmdeploy/vl/model/phi3_vision.py]: could be removed

File: lmdeploy/archs.py (hunk 1 above)
Comment (RunningLeon) [reply lmdeploy/archs.py]: done

**Review by lvhan028 — APPROVED** [review]

**Review by AllentDan — APPROVED** [review]
Tested OK with api_server

**Pull request merged**